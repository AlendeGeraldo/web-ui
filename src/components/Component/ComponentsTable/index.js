/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';

import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Popconfirm, Table } from 'antd';

import './style.scss';

// eslint-disable-next-line
class ComponentTable extends React.Component {
  render() {
    const { componentList, onDelete } = this.props;

    const tableColumns = [
      {
        title: 'Nome do Componente',
        dataIndex: 'name',
        key: 'name',
        render: (value, record) => (
          <Link to={`components/${record.uuid}`}>{value}</Link>
        ),
        sorter: (a, b) => a.name.localeCompare(b.name),
        ellipsis: true,
      },
      {
        title: 'Data de Criação',
        dataIndex: 'createdAt',
        key: 'createdAt',
        width: 200,
        render: (value) => new Date(value).toLocaleString(),
        sorter: (a, b) => new Date(a.createdAt) - new Date(b.createdAt),
        ellipsis: true,
      },
      {
        title: '',
        key: 'action',
        width: 100,
        render: (record) => (
          <Popconfirm
            value={record}
            title={`Tem certeza de que deseja excluir o componente ${record.name} ?`}
            onConfirm={() => {
              onDelete(record);
            }}
            okText='Sim'
            cancelText='Não'
            placement='left'
          >
            <a href='#'>Remover</a>
          </Popconfirm>
        ),
      },
    ];

    return (
      <Table
        className='componentsTable'
        rowKey={(record) => record.uuid}
        dataSource={componentList}
        columns={tableColumns}
        pagination={{ pageSize: 9 }}
        scroll={{ y: 'calc(100vh - 480px)' }}
      />
    );
  }
}

ComponentTable.propTypes = {
  componentList: PropTypes.arrayOf(PropTypes.object).isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default ComponentTable;
