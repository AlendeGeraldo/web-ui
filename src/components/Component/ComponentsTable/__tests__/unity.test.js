import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';
import { Popconfirm, Table } from 'antd';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import ComponentsTable from '..';

const components = [
  {
    uuid: '9014c0e6-534b-4e7d-a8db-cd5c9d8ee540',
    createdAt: '2019-11-05T17:16:53.000Z',
    updatedAt: null,
    name: 'teste 2',
    file: '',
  },
  {
    uuid: '58af2581-de59-40ea-8051-cf28a9c3f5b1',
    createdAt: '2019-11-04T17:16:48.000Z',
    updatedAt: null,
    name: 'teste 1',
    file: '',
  },
  {
    uuid: '80b35439-d45b-4ae1-b437-8069a5663510',
    createdAt: '2019-11-07T17:16:43.000Z',
    updatedAt: null,
    name: 'teste ',
    file: '',
  },
];

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  components: {
    componentList: components,
  },
});

const setupShallow = () => {
  const wrapper = shallow(<ComponentsTable store={store} />);
  return wrapper.dive().dive();
};

const setupMount = () => {
  const wrapper = mount(
    <Provider store={store}>
      <Router>
        <ComponentsTable />
      </Router>
    </Provider>
  );
  return wrapper;
};

describe('ComponentTable component', () => {
  it('is expected render without crashing', () => {
    setupShallow();
  });

  it('is expected render html correctly', () => {
    const wrapper = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('is expected to exist Table', () => {
    const wrapper = setupShallow();
    expect(wrapper.find(Table).exists()).toBeTruthy();
  });

  it('sort records', () => {
    const wrapper = setupMount();

    const renderedNames = () => {
      return wrapper.find('TableRow').map((row) => row.props().record.name);
    };

    const sorters = wrapper.find('.ant-table-column-sorters');

    // ascend name
    sorters.at(0).simulate('click');
    expect(renderedNames()).toEqual(['teste ', 'teste 1', 'teste 2']);

    // descend name
    sorters.at(0).simulate('click');
    expect(renderedNames()).toEqual(['teste 2', 'teste 1', 'teste ']);

    // ascend create date
    sorters.at(1).simulate('click');
    expect(renderedNames()).toEqual(['teste 1', 'teste 2', 'teste ']);

    // descend create date
    sorters.at(1).simulate('click');
    expect(renderedNames()).toEqual(['teste ', 'teste 2', 'teste 1']);
  });

  it('onDeleteComponent should be called', () => {
    const wrapper = setupMount();
    const popconfirm = wrapper.find(Popconfirm);
    const triggerNode = popconfirm.at(0).find('[href="#"]');
    triggerNode.simulate('click');
    expect(wrapper.find('.ant-btn-primary').simulate('click')).toBeTruthy();
  });
});
