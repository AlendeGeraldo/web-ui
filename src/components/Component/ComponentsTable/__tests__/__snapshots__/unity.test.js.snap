// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTable component is expected render html correctly 1`] = `
<withStore(Table)
  className="componentsTable"
  columns={
    Array [
      Object {
        "dataIndex": "name",
        "ellipsis": true,
        "key": "name",
        "render": [Function],
        "sorter": [Function],
        "title": "Nome do Componente",
      },
      Object {
        "dataIndex": "createdAt",
        "ellipsis": true,
        "key": "createdAt",
        "render": [Function],
        "sorter": [Function],
        "title": "Data de Criação",
        "width": 200,
      },
      Object {
        "key": "action",
        "render": [Function],
        "title": "",
        "width": 100,
      },
    ]
  }
  dataSource={
    Array [
      Object {
        "createdAt": "2019-11-05T17:16:53.000Z",
        "file": "",
        "name": "teste 2",
        "updatedAt": null,
        "uuid": "9014c0e6-534b-4e7d-a8db-cd5c9d8ee540",
      },
      Object {
        "createdAt": "2019-11-04T17:16:48.000Z",
        "file": "",
        "name": "teste 1",
        "updatedAt": null,
        "uuid": "58af2581-de59-40ea-8051-cf28a9c3f5b1",
      },
      Object {
        "createdAt": "2019-11-07T17:16:43.000Z",
        "file": "",
        "name": "teste ",
        "updatedAt": null,
        "uuid": "80b35439-d45b-4ae1-b437-8069a5663510",
      },
    ]
  }
  pagination={
    Object {
      "pageSize": 9,
    }
  }
  rowKey={[Function]}
  scroll={
    Object {
      "y": "calc(100vh - 480px)",
    }
  }
/>
`;
